src/room_member.cpp

```diff
@@ -256,12 +256,24 @@ bool RoomMember::IsConnected() const {
     return state == State::Joining || state == State::Joined;
 }
 
+void RoomMember::CloseConnection() {
+    peer->Shutdown(300);
+    member_information.clear();
+    nickname.clear();
+    mac_address = {};
+    room_information = {};
+
+    chat_queue.clear();
```

Comment:
Yeah, even after leaving the queues should be maintained so we can receive all remaining messages